ludwig/utils/data_utils.py

```diff
@@ -299,6 +300,26 @@ def read_hdf5(data_fp, **kwargs):
     return load_hdf5(data_fp, clean_cols=True)
 
 
+@DeveloperAPI
+@spread
+def read_buffer(buf, fname):
+    data_format = figure_data_format_dataset(fname)
+    reader_fn = data_reader_registry[data_format]
+    with tempfile.TemporaryDirectory() as tmpdir:
+        temp_name = os.path.join(tmpdir, "dataset")
+        with open(temp_name, "wb") as f:
+            f.write(buf.read())
+        return reader_fn(temp_name, pd)
+
+
+@DeveloperAPI
+@spread
+def read_fname(fname, data_format=None, df_lib=pd, **kwargs):
```

Comment:
Can we add the descriptions from the PR description as docstrings to these functions?

May also be useful to include a comment on why or when these functions would be useful, over existing `read_*` functions.

Reply:
Addressed feedback!